### PR TITLE
Added Delete API and Update Service to allow Editing from UI

### PR DESCRIPTION
**Pull request by deepaksingh1308**
- Added Delete API
- Refactored Model to allow edit / delete
- Updated Service Methods for APIs accordingly
- Removed some unused imports
- Merged with Global Search API branch

### CODE REVIEW

File: src/main/java/com/wf/stp/rbms/domain/CreateRulesRequest.java [hunk 1]
```diff
@@ -11,10 +11,12 @@
 @AllArgsConstructor
 public class CreateRulesRequest {
 
+    private String id;
     private String serviceName;
     private String ruleName;
     private String ruleDescription;
     private String priority;
+    private Boolean isEdited;
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/domain/CreateRulesRequest.java:19]: Please use a primitive 'boolean' instead.

File: src/main/java/com/wf/stp/rbms/service/UpoCreateRulesService.java [hunk 2]
```diff
@@ -98,30 +108,50 @@ public UPORuleResponse findRuleById(String id) {
...
+        query.addCriteria(Criteria.where("_id").is(id));
+
+        final Update update = new Update();
+        update.set(IS_ACTIVE, false);
+        update.set("endTimeStamp", LocalDateTime.now().toString());
+
+        return mongoTemplate.updateFirst(query, update, DroolsRuleModel.class).wasAcknowledged();
+    }
+
+    public List<UPORuleNameResponse> findRulesBySubstring(String ruleNameSbstr) {
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/service/UpoCreateRulesService.java:130]: Please rename this to findRulesByCriteria(String criteria) instead.

**Review by rkpandita — APPROVED** [review]
Looks good.

**Review by YogeshPitale — APPROVED** [review]
Looks good!

**Pull request merged**